### PR TITLE
Remove uninstalled plugin from plugin manager

### LINKED ISSUE #4937: plugin installer bugs
## 🐛 Bug
looks like the plugin installer has accumulated a number of bugs:

- [ ] when uninstalling an npe2 plugin, the warning that pops up has an unclickable "close" button

    https://user-images.githubusercontent.com/1609449/184726352-8321dd46-f326-4fa4-bb74-f25112b769ef.mov

- [ ] When you install a plugin, it no longer moves from the "available" section to the "installed" section (though it _does_ get installed)

   https://user-images.githubusercontent.com/1609449/184726452-d3cc72d3-07eb-4521-a2e5-7a5bfa781490.mov

- [ ] When you uninstall something the available list goes into a permanent clear/refresh loop that needs a restart to fix

    https://user-images.githubusercontent.com/1609449/184726526-8756f0fd-3d48-4902-9268-688b6ec9bcca.mov

### LINKED ISSUE #5188: Plugin installation dialog "hang" after uninstall
## 🐛 Bug

After uninstalling a plugin, the installation dialog keeps refreshing and effectively prevents any action from happening.

## To Reproduce

1. install a plugin from napari plugin dialog (optional)
2. uninstall the plugin installed from previous step
3. see the plugin dialog keeps refreshing

## Expected behavior

The plugin dialog should show the latest status instead of hanging in refreshing

## Environment

 - main branch napari or 0.4.17 rc both have this behavior
 - this is discovered on macOS, not sure if can be reproduced elsewhere

### PR DESCRIPTION
**Pull request by Czaki**
# Description

Bugfix of #5188

In this PR I removed the plugin scheduled for uninstalling from npe2 plugin manager

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

closes #4937
closes #5188

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

@potating-potato @psobolewskiPhD 
The the core bug is fixed, but I still have a problem with the closing dialog. Could You check how it works for you?

### CODE REVIEW

File: napari/_qt/dialogs/qt_plugin_dialog.py [hunk 1]
```diff
@@ -816,6 +819,7 @@ def _add_to_installed(distname, enabled, npe_version=1):
             )
 
         pm2 = PluginManager.instance()
+        pm2.discover()
```
Comment (Czaki) [inline napari/_qt/dialogs/qt_plugin_dialog.py]: this line is to show the freshly installed plugins in the upper list. But still, installing npe2 does not cause popup with information that reinstallation is required.

**Comment by psobolewskiPhD** [comment]
Edit: not actually on the right branch, disregard.

File: napari/_qt/dialogs/qt_plugin_dialog.py (hunk 1 above)
Comment (psobolewskiPhD) [reply napari/_qt/dialogs/qt_plugin_dialog.py]: Edit: Yup, works correctly now!

**Comment by Czaki** [comment]
> Also I'm not sure if it's related to this PR, but I note an extra issue: when installing an `npe2` (you have no way of knowing, when you do at the moment) it doesn't show up in the installed list, just in available with the install icon. Show status shows that it did install and restarting napari shows it correctly, but I thought there was a warning that should pop for that?

It looks like you do not test this branch. This is also fixed here. I even add a proper comment to the code.

**Comment by psobolewskiPhD** [comment]
> It looks like you do not test this branch. This is also fixed here. I even add a proper comment to the code.

Christ, you're right. Sorry I'm so useless. I think I have it correctly installed now.
Edit: Ok, uninstall npe2 is perfectly fine now, napari functions normally, other plugins work.
Edit2: Installing npe2 is also fixed. it properly shows up in the Installed.

So as far As I can tell this PR fixes: https://github.com/napari/napari/issues/4937

**Comment by Czaki** [comment]
> Christ, you're right. Sorry I'm so useless. I think I have it correctly installed now.

No. You are just unfamiliar with GIT. Git is powerful but sometimes hard, and sometimes learning is pain (from my experience). 

I just added a fix for hard to click close button

**Comment by psobolewskiPhD** [comment]
> I just added a fix for hard to click close button

Yup! Works perfectly now.
Edit: I can sit and install/uninstall the same plugin and it all works. Closing out the plugin GUI leaves napari normal.

**Comment by Czaki** [comment]
The core problem with the close button is that this button was below the text, and only its left and bottom were clickable. 

I just moved it above the label.

**Comment by Czaki** [comment]
I have added https://github.com/napari/npe2/pull/248 PR that will allow in the future to show information that there is a need to restart napari after installing a npe2 plugin.

**Comment by psobolewskiPhD** [comment]
> I have added [napari/npe2#248](https://github.com/napari/npe2/pull/248) PR that will allow in the future to show information that there is a need to restart napari after installing a npe2 plugin.

I'll try to test this later or tomorrow. 
Should I just fork npe2 and `pip install -e .` into my napari-dev environment (the `.[dev]` one)? 
Or is there a better way you recommend?

**Comment by Czaki** [comment]
> I'll try to test this later or tomorrow.
> Should I just fork npe2 and `pip install -e .` into my napari-dev environment (the `.[dev]` one)?
> Or is there a better way you recommend?

I first need to update this PR to be compatible with this update npe2.

**Comment by Czaki** [comment]
Update landed. to test it, you need to install npe2 from PR (and maybe ignore the warning about the incompatible version if the tags were not correctly handled - message screenshot bellow). 

obraz

It should be backward compatible as the `discover` method currently returns `None` that evaluates to false.

**Comment by andy-sweet** [comment]
@goanpeca : I assigned you as a reviewer as you seem quite familiar with this area of the code. Feel free to unassign yourself or assign someone else if you can't get to it in the next few days.

**Comment by psobolewskiPhD** [comment]
@Czaki I forked and installed the npe2 PR plus this PR and it works nicely for me!
The little warning pops up in the middle of the screen.

Edit: when I checkout napari/main obv. the warning doesn't pop up, but otherwise things behave normally.
And if I checkout this PR but npe2/main also the warning doesn't pop, but everything else works.
If that's what you mean about backwards compatibility....

**Comment by Czaki** [comment]
> And if I checkout this PR but npe2/main also the warning doesn't pop, but everything else works.

yes. This is what I mean. So This PR does not require new npe2 release but will have benefits from it.

**Review by goanpeca — APPROVED** [review]
Looks good @Czaki !

Thanks for working on this! 🚀

**Pull request merged**